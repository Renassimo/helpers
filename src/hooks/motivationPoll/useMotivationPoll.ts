import { useContext } from 'react';

import MotivationPollContext from 'src/contexts/motivationPoll';

const useMotivationPoll = () => useContext(MotivationPollContext);

export default useMotivationPoll;
