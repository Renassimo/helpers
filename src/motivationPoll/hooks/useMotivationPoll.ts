import { useContext } from 'react';

import MotivationPollContext from '@/motivationPoll/contexts/MotivationPollContext';

const useMotivationPoll = () => useContext(MotivationPollContext);

export default useMotivationPoll;
