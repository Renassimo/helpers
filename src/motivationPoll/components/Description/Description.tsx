import useMotivationPoll from '@/hooks/motivationPoll/useMotivationPoll';

import Typography from '@mui/material/Typography';
import Box from '@mui/material/Box';

const Description = () => {
  const { description } = useMotivationPoll();
  return (
    <Box>
      <Typography component="h1" variant="h4" textAlign="center" m={2}>
        Motivation Test
      </Typography>
      <Typography component="p" m={4}>
        {description}
      </Typography>
    </Box>
  );
};

export default Description;
