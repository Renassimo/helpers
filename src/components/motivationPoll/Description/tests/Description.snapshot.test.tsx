import renderWithTheme from '@/tests/helpers';
import useMotivationPoll from '@/hooks/motivationPoll/useMotivationPoll';
import Description from '../Description';

jest.mock('@/hooks/motivationPoll/useMotivationPoll');

describe('Description snapshot', () => {
  const mockedDescription = 'Mocked description';

  beforeEach(() => {
    (useMotivationPoll as jest.Mock).mockImplementation(() => ({
      description: mockedDescription,
    }));
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders successfully', () => {
    // Arange
    // Act
    const { baseElement } = renderWithTheme(<Description />);
    // Assert
    expect(baseElement).toMatchSnapshot();
  });
});
