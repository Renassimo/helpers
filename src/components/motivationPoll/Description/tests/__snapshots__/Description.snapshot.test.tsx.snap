// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description snapshot renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <h1
        class="MuiTypography-root MuiTypography-h4 css-1nxg358-MuiTypography-root"
      >
        Motivation Test
      </h1>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1jmylmz-MuiTypography-root"
      >
        Mocked description
      </p>
    </div>
  </div>
</body>
`;
